src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private RandomSize size = new RandomSize();
+
+    private Figure circle = new Circle(size.getSize(), colorSupplier.getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(size.getSize(),
+            size.getSize(), size.getSize(), colorSupplier.getRandomColor());
+    private Figure rectangle = new Rectangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+    private Figure square = new Circle(size.getSize(),
+            colorSupplier.getRandomColor());
+    private Figure rightTriangle = new RightTriangle(size.getSize(),
+            size.getSize(), colorSupplier.getRandomColor());
+
+    private Figure [] figures = {
+            circle,
+            isoscelesTrapezoid,
+            rectangle,
+            square,
+            rightTriangle
+    };
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(figures.length);
+        return figures[randomIndex];
+    }
+
+    public Figure getDefaultFigure() {
+        Figure defaulFigure = new Circle(10, Color.WHITE);

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. Also, there's a typo in the variable name 'defaulFigure', it should be 'defaultFigure'.